src/v/pandaproxy/kafka_client_cache.h

```diff
@@ -13,6 +13,9 @@
 #include "config/rest_authn_endpoint.h"
 #include "pandaproxy/types.h"
 
+#include <seastar/core/gate.hh>
```

Comment:
`gate` doesn't seem to be used, but I think it should be. Stopping the clients is asynchronous, and a gate should be held so that it can be waiting upon during `stop`, so that it's not used after destruction.

Is this duplicating some of the functionality in `sharded_client_cache` now? Would it be clearer to move that code rather than duplicate it?

Reply:
> Is this duplicating some of the functionality in `sharded_client_cache` now? Would it be clearer to move that code rather than duplicate it?

Yes that would be more convenient. I'll look into merging sharded_client_cache and kafka_client_cache